src/components/dialogs/equipment-id/equipment-id-selector.js

```diff
@@ -67,7 +67,11 @@ export const EquipmentIdSelector = ({
     }, [selectedValue, setSelectedId]);
 
     const handleChange = (newId, reason) => {
-        if (newId && (reason === 'createOption' || reason === 'selectOption')) {
+        if (
+            newId &&
+            ['createOption', 'selectOption'].includes(reason) &&
+            equipmentOptions.includes(newId)
```

Comment:
`equipmentOptions.includes(newId)` breaks the need to validate a non-existing ID